substrate/frame/support/procedural/src/construct_runtime_v2/expand/mod.rs

```diff
@@ -0,0 +1,289 @@
+				)
+			}),
+	};
+
+	let res = res.unwrap_or_else(|e| e.to_compile_error());
+
+	let res = expander::Expander::new("construct_runtime")
+		.dry(std::env::var("FRAME_EXPAND").is_err())
+		.verbose(true)
+		.write_to_out_dir(res)
+		.expect("Does not fail because of IO in OUT_DIR; qed");
+
+	res.into()
+}
+
+fn construct_runtime_implicit_to_explicit(
+	input: TokenStream2,
+	definition: ImplicitAllPalletsDeclaration,
+) -> Result<TokenStream2> {
+	let frame_support = generate_crate_access_2018("frame-support")?;
+	let mut expansion = quote::quote!(
+		#[frame_support::construct_runtime_v2]
+		#input
+	);
+	for pallet in definition.pallet_decls.iter() {
+		let pallet_path = &pallet.path;
+		let pallet_name = &pallet.name;
+		let pallet_instance = pallet.instance.as_ref().map(|instance| quote::quote!(<#instance>));
+		expansion = quote::quote!(
+			#frame_support::__private::tt_call! {
+				macro = [{ #pallet_path::tt_default_parts_v2 }]
+				frame_support = [{ #frame_support }]
+				~~> #frame_support::match_and_insert! {
+					target = [{ #expansion }]
+					pattern = [{ #pallet_name: #pallet_path #pallet_instance  }]
+				}
+			}
+		);
+	}
+
+	Ok(expansion)
+}
+
+fn construct_runtime_final_expansion(
+	name: Ident,
+	definition: ExplicitAllPalletsDeclaration,
+	runtime_types: Vec<RuntimeType>,
+) -> Result<TokenStream2> {
+	let ExplicitAllPalletsDeclaration { pallets, name: pallets_name } = definition;
+
+	let system_pallet =
+		pallets.iter().find(|decl| decl.name == SYSTEM_PALLET_NAME).ok_or_else(|| {
+			syn::Error::new(
+				pallets_name.span(),
+				"`System` pallet declaration is missing. \
+			 Please add this line: `System: frame_system,`",
+			)
+		})?;
+	if !system_pallet.cfg_pattern.is_empty() {
+		return Err(syn::Error::new(
+			system_pallet.name.span(),
+			"`System` pallet declaration is feature gated, please remove any `#[cfg]` attributes",
+		))
+	}
+
+	let features = pallets
+		.iter()
+		.filter_map(|decl| {
+			(!decl.cfg_pattern.is_empty()).then(|| {
+				decl.cfg_pattern.iter().flat_map(|attr| {
+					attr.predicates().filter_map(|pred| match pred {
+						Predicate::Feature(feat) => Some(feat),
+						Predicate::Test => Some("test"),
+						_ => None,
+					})
+				})
+			})
+		})
+		.flatten()
+		.collect::<HashSet<_>>();
+
+	let hidden_crate_name = "construct_runtime";
+	let scrate = generate_crate_access(hidden_crate_name, "frame-support");
+	let scrate_decl = generate_hidden_includes(hidden_crate_name, "frame-support");
+
+	let frame_system = generate_crate_access_2018("frame-system")?;
+	let block = quote!(<#name as #frame_system::Config>::Block);
+	let unchecked_extrinsic = quote!(<#block as #scrate::sp_runtime::traits::Block>::Extrinsic);
+
+	let mut dispatch = quote!();
+	let mut outer_event = quote!();
+	let mut outer_error = quote!();
+	let mut outer_origin = quote!();
+	let mut freeze_reason = quote!();
+	let mut hold_reason = quote!();
+	let mut slash_reason = quote!();
+	let mut lock_id = quote!();
```

Comment:
I would feel a little bit better if these were `Option`-based so we know later in the code which have been specified and which haven't... right now only way to tell is `TokenStream::is_empty()` so we're basically bypassing rust's option semantics using blank quotes.

Could easily just promote these to `Option<TokenStream2>` and they will function the same way they do now in the final quote